src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -46,12 +48,18 @@
 
     private RemoteInvokerFactory remoteInvokerFactory = RemoteInvokerFactory.DEFAULT;
     private Function<Client, Client> decorator;
+    private String groupName;
 
     /**
      * Creates a new {@link ClientBuilder} that builds the client that connects to the specified {@code uri}.
      */
     public ClientBuilder(String uri) {
         this(URI.create(requireNonNull(uri, "uri")));
+
+        if (this.uri.getAuthority().equals(this.uri.getHost()) &&
+                ServerGroupRegistry.get(this.uri.getHost()) != null) {
+            this.groupName = this.uri.getHost();
```

Comment:
As @anuraaga said before, using full hostname as a group name can confuse a user. Could we define some naming rule instead?

My suggestion is using the authority section of the URI like the following:

    http://serverGroup:myGroupName:8080/
    http://username:password@serverGroup:myGroupName:8080/

If the 'host' part of the URI starts with `serverGroup:`, we could safely consider that `myGroupName` is a `ServerGroup` name.

@shelocks @anuraaga WDYT?